pkg/apis/environment/extension_view.go

```diff
@@ -27,3 +33,91 @@ type (
 		Edges    []GraphEdge   `json:"edges"`
 	}
 )
+
+type (
+	RouteCloneEnvironmentRequest struct {
+		_ struct{} `route:"POST=/clone"`
+
+		CloneEnvironmentId object.ID `path:"environment"`
+
+		model.EnvironmentCreateInput `path:",inline" json:",inline"`
+	}
+
+	RouteCloneEnvironmentResponse = model.EnvironmentOutput
+)
+
+func (r *RouteCloneEnvironmentRequest) Validate() error {
```

Comment:
It seems that the clone request validate is similar to the create environment validate, can you using a common validate func for two request validation?

Reply:
updated